tests/API1/testparameterizedobject.py

```diff
@@ -591,7 +591,8 @@ class B(A):
 
     b = B()
 
-    assert b.p == 0.0
+    # Could argue this should not be allowed.
+    assert b.p == '1'
```

Comment:
I spent a lot of time adding tests to *main*, in an attempt to capture these sorts of change in behavior. I think that Param has been out for long enough than anything it does should be considered a feature, and changing any of that is effectively a breaking change. Breaking stuff is fine, but the more of these changes we can capture, the better we can gauge how it is going to affect users and analyze whether the change is worth it.

> because changing String to Number in a subclass is a semantic error

I agree that it is not something users should do. However, there is at this stage nothing in Param against that:
- I don't think the docs mention that you shouldn't do that
- Metadata inheritance does not check for any sort of relationship between Parameters

Python itself happily lets you happily override an attribute in a subclass with anything you want. Some other languages don't?

I suggest that we improve the docs by saying that Parameter metadata is inherited without any semantic check, and we suggest to write classes that do respect that to avoid subtle bugs (e.g. `A` defines `param.Number`, `B` specifies `param.Integer`)

> unless someone objects

I guess you meant someone that is not me, I nominate @jlstevens !